lib/decorate.js

```diff
@@ -56,9 +56,19 @@ function decorateConstructor (konstructor, name, fn, dependencies) {
   }
 }
 
-function checkReferenceType (name, fn) {
+function encapsulateReferenceType (name, fn) {
   if (typeof fn === 'object' && fn && !(typeof fn.getter === 'function' || typeof fn.setter === 'function')) {
-    FSTDEP006(name)
+    return encapsulatedDecorator(fn)
```

Comment:
I would just throw in this case and not clone.

Reply:
So are we removing the warning to replace it with an error while suggesting the user use the getter/setter interface?

Reply:
Yes.

Reply:
Done